src/js/components/award/shared/activity/ContractGrantActivityChart.jsx

```diff
@@ -40,34 +42,26 @@ const ContractGrantsActivityChart = ({
     const [xTicks, setXTicks] = useState([]);
     // y ticks
     const [yTicks, setYTicks] = useState([]);
+    // start line
+    const [startLineValue, setStartLineValue] = useState(null);
+    // end line
+    const [endLineValue, setEndLineValue] = useState(null);
+    // potential end line
+    const [potentialEndLineValue, setPotentialEndLineValue] = useState(null);
+    // x axis spacing
+    const [xAxisSpacing, setXAxisSpacing] = useState(0);
+    // circle data
+    const [circleData, setCircleData] = useState([]);
+    // area path
+    const [areaPath, setAreaPath] = useState('');
     /**
```

Comment:
Could we ultimately make this into a reusable hook like here? If not, why not?

https://github.com/fedspendingtransparency/usaspending-website/blob/dev/src/js/components/award/shared/awardAmountsSection/charts/AwardAmountsChart.jsx#L20

Reply:
I definitely do not think I understand?

Reply:
> How do we test functional containers?

Oh man... this is a problem.